### PR TITLE
Add sigmoid op based on MKL-DNN

### PR DESCRIPTION
**Pull request by wuhuikx**

### CODE REVIEW

**Comment by wuhuikx** [comment]
@yinghai @jgong5

**Review by yinghai — APPROVED** [review]
LGTM

**Review by yinghai — CHANGES_REQUESTED** [review]
I saw some errors in CI. Could you rebase?

File: caffe2/ideep/operators/sigmoid_op.cc [hunk 1]
```diff
@@ -0,0 +1,60 @@
+#include <caffe2/ideep/ideep_utils.h>
+
+namespace caffe2 {
+
+class IDEEPSigmoidOp final : public IDEEPOperator {
```
Comment (yinghai) [inline caffe2/ideep/operators/sigmoid_op.cc]: Please put these classes in anonymous namespace.

File: caffe2/python/ideep/sigmoid_op_test.py [hunk 2]
```diff
@@ -0,0 +1,30 @@
...
+from __future__ import print_function
+from __future__ import unicode_literals
+
+import unittest
+import hypothesis.strategies as st
+from hypothesis import given
+import numpy as np
+from caffe2.python import core, workspace
+import caffe2.python.hypothesis_test_util as hu
+import caffe2.python.ideep_test_util as mu
```
Comment (yinghai) [inline caffe2/python/ideep/sigmoid_op_test.py]: unused.

File: caffe2/python/ideep/sigmoid_op_test.py [hunk 3]
```diff
@@ -0,0 +1,30 @@
...
+
+@unittest.skipIf(not workspace.C.use_ideep, "No IDEEP support.")
```
Comment (yinghai) [inline caffe2/python/ideep/sigmoid_op_test.py]: Two empty lines instead of line

File: caffe2/python/ideep/sigmoid_op_test.py [hunk 4]
```diff
@@ -0,0 +1,30 @@
...
+        op = core.CreateOperator(
+            "Sigmoid",
+            ["X"],
+            ["Y"] if not inplace else ["X"],
+        )
+
+        self.assertDeviceChecks(dc, op, [X], [0])
+        self.assertGradientChecks(gc, op, [X], 0, [0])
+
+if __name__ == "__main__":
```
Comment (yinghai) [inline caffe2/python/ideep/sigmoid_op_test.py]: Two empty lines instead of line

**Review by yinghai — COMMENTED** [review]
Could you rebase and see if the CI errors are gone?

File: caffe2/ideep/operators/sigmoid_op.cc (hunk 1 above)
Comment (wuhuikx) [reply caffe2/ideep/operators/sigmoid_op.cc]: Hi, I found the other ops (e.g. caffe2/ideep/operators/pool_op.cc) are all put the classes in caffe2 namespace. May I know why this sigmoidop should be in anonymous namespace?

**Review by yinghai — APPROVED** [review]

**Pull request closed without merging**